src/components/content/heroes/aurora/aurora.tsx

```diff
@@ -1,62 +1,52 @@
-import styled from 'styled-components';
-
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getWeather } from './weather-api';
 
 import AuroraBlurredBackgroundA from '../../../../assets/images/aurora/bg-blur-a.png';
 import AuroraBlurredBackgroundB from '../../../../assets/images/aurora/bg-blur-b.png';
 import AuroraBlurredBackgroundC from '../../../../assets/images/aurora/bg-blur-c.png';
 
 import { Flare, FlareType } from './flare';
-
-const BACKGROUND_LAYER_Z = -1;
-const FOREGROUND_LAYER_Z = 0;
-
-export enum AuroraType {
-  Pink = 'pink',
-  Blue = 'blue',
-  BrightBlue = 'bright-blue',
-}
-
-interface AuroraBackgroundProps {
-  source: string;
-}
-
-const AuroraBackground = styled.div<AuroraBackgroundProps>`
-  background-color: #202840;
-  position: absolute;
-
-  background-image: url(${(props) => props.source});
-  background-repeat: no-repeat;
-  background-size: cover;
-  background-position: center -20vw;
-  z-index: ${BACKGROUND_LAYER_Z};
-  left: 0;
-  right: 0;
-  bottom: 0;
-  top: 0;
-`;
-
-const AuroraForeground = styled.div`
-  position: absolute;
-  z-index: ${FOREGROUND_LAYER_Z};
-  overflow: hidden;
-  left: 0;
-  right: 0;
-  bottom: 0;
-  top: 0;
-`;
-
-const AuroraContainer = styled.div`
-  pointer-events: none;
-`;
+import { AuroraType, WeatherType } from './aurora-types';
+import {
+  flaresByWeather,
+  AuroraForeground,
+  AuroraContainer,
+  AuroraBackground,
+} from './aurora-components';
 
 export interface AuroraProps {
   type?: AuroraType;
   className?: string;
 }
 
 export const Aurora = ({ type, className }: AuroraProps) => {
+  const [weather, setWeather] = useState(WeatherType.NotSet);
+
+  const toggleWeather = async () => {
+    if (weather === WeatherType.NotSet) {
+      setWeather(await getWeather());
+    } else {
+      setWeather(WeatherType.NotSet);
+    }
+  };
+
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.ctrlKey && event.altKey && event.shiftKey) {
+        toggleWeather().then();
+      }
+    };
+
+    document.addEventListener('keydown', handleKeyDown);
+
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  });
   function getSource(type?: AuroraType) {
+    if (weather !== WeatherType.NotSet) {
+      return '';
+    }
```

Comment:
Die Logik würde ich eher direkt in `AuroraBackground` schreiben. Also das `background-image` in `AuroraBackground` mit einer Bedingung setzten.